src/screens/nano/NanoContractDetail.js

```diff
@@ -0,0 +1,203 @@
+import { addBlueprintInformation } from '../../actions';
+import { get } from 'lodash';
+import { GlobalModalContext, MODAL_TYPES } from '../../components/GlobalModal';
+
+
+/**
+ * Details of a Nano Contract
+ *
+ * @memberof Screens
+ */
+function NanoContractDetail(props) {
+  const context = useContext(GlobalModalContext);
+
+  const { nanoContracts, blueprintsData, tokenMetadata } = useSelector((state) => {
+    return {
+      nanoContracts: state.nanoContracts,
+      blueprintsData: state.blueprintsData,
+      tokenMetadata: state.tokenMetadata
+    }
+  });
+
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const { nc_id: ncId } = useParams();
+  const nc = nanoContracts[ncId];
+  let blueprintInformationAux = blueprintsData[nc.blueprintId];
+
+  // loading {boolean} Bool to show/hide loading element
+  const [loading, setLoading] = useState(true);
+  // data {Object} Nano contract loaded data
+  const [data, setData] = useState(null);
+  // blueprintInformation {Object} Blueprint information data
+  const [blueprintInformation, setBlueprintInformation] = useState(blueprintInformationAux);
+  // errorMessage {string} Message to show when error happens on the form
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    loadData();
+  }, []);
+
+  /**
+   * Method executed when link to execute a method is clicked
+   *
+   * @param {Object} e Event emitted by the link clicked
+   * @param {String} method Method to be executed
+   */
+  const executeMethod = (e, method) => {
+    e.preventDefault();
+    navigate('/nano_contract/execute_method/', {
+      state: {
+        method,
+        ncId,
+      },
+    });
+  }
+
+  /**
+   * Method executed when link to change the address is clicked
+   *
+   * @param {Object} e Event emitted by the link clicked
+   */
+  const changeAddress = (e) => {
+    e.preventDefault();
+    context.showModal(MODAL_TYPES.NANOCONTRACT_CHANGE_ADDRESS, {
+      nanoContractID: ncId
+    });
+  }
+
+  const loadData = async () => {
+    await loadBlueprintInformation();
+    await loadNCData();
+  }
+
+  const loadBlueprintInformation = async () => {
+    if (blueprintInformationAux) {
+      return;
+    }
+
+    setLoading(true);
+    try {
+      const blueprintInformationResponse = await hathorLib.ncApi.getBlueprintInformation(nc.blueprintId);
+      // We need this blueprint information response to call the following get state
+      // The set state is not sync, so we need to store it in a common variable to be used in the next call
+      blueprintInformationAux = blueprintInformationResponse;
+      setBlueprintInformation(blueprintInformationResponse);
+      // Store in redux, so it can be reused by other nano contracts
+      dispatch(addBlueprintInformation(blueprintInformationResponse));
+    } catch(e) {
+      // Error in request
+      setLoading(false);
+      setErrorMessage(t`Error getting blueprint details.`);
+    };
+  }
```

Comment:
If an error is not thrown, who sets the loading to false?
Is this working because loadNCData is called right after?

If so, why not move the `setLoading` pair to the `loadData` method?

Reply:
You are correct regarding `setLoading(false)`.

The method `loadBlueprintInformation` might not be called, in case this blueprint information is already loaded but if it's called, the loading must start in the beginning.

So `setLoading(true)` should exist in both methods but `setLoading(false)` only in the second.

Removed the one from the first method. https://github.com/HathorNetwork/hathor-wallet/pull/623/commits/76798852607b4f1d6867f088a306a125049cced1